saito-core/src/common/test_manager.rs

```diff
@@ -537,17 +572,157 @@ pub mod test {
             GoldenTicket::new(block_hash, random_bytes, public_key)
         }
 
+        pub fn get_mempool_lock(&self) -> Arc<RwLock<Mempool>> {
+            return self.mempool_lock.clone();
+        }
+
+        pub fn get_wallet_lock(&self) -> Arc<RwLock<Wallet>> {
+            return self.wallet_lock.clone();
+        }
+
+        pub async fn get_wallet(&self) -> tokio::sync::RwLockReadGuard<'_, Wallet> {
+            let wallet;
+            let _wallet_;
+            (wallet, _wallet_) = lock_for_read!(self.wallet_lock, LOCK_ORDER_WALLET);
+            return wallet;
```

Comment:
this shouldn't be returned like this. until returned wallet is destroyed lock is kept.